lib/model/narrow.dart

```diff
@@ -260,11 +265,11 @@ class DmNarrow extends Narrow implements SendableNarrow {
   late final String _key = otherRecipientIds.join(',');
 
   @override
-  bool containsMessage(Message message) {
-    if (message is! DmMessage) return false;
-    if (message.allRecipientIds.length != allRecipientIds.length) return false;
+  bool containsMessage(MessageBase message) {
+    if (message is! MessageBase<DmConversation>) return false;
+    if (message.conversation.allRecipientIds.length != allRecipientIds.length) return false;
```

Comment:
This is probably a bit cleaner with a `conversation` local, like the ChannelNarrow and TopicNarrow implementations above.